site/app/templates/notebook/Notebook.twig

```diff
@@ -262,6 +271,41 @@
 </div>
 
 <script>
+    //for displaying popup submit button
+    //interval is 20 percent of overall time allowed or 10 minutes, whichever is greater
+    let interval = Math.floor(((({{ allowed_minutes }}) / 5) > 10 ? ({{ allowed_minutes }}) / 5 : 10));
+    let loadTime = 0;
+    let showPopup = false;
+
+    function initializePopupTimer(){
```

Comment:
When does this function fire? Always on the first time we open the submission page?

I'm trying to understand when the localStorage is getting set, and if it could be replaced with a local variable.

Reply:
yes it fires the first time the user goes to the submission page, I used session storage so that the variable doesn't persist for a different gradeable.

Reply:
I'm not sure `sessionStorage` works the way you think it does?

For my understanding, let me know if the following is correct (with all navigation happening in the same page):
* User goes to notebook gradeable A, `initializePopupTimer` fires
* User navigates to a different page
* User goes to notebook gradeable A, `initializePopupTimer` fires
* User goes to a different page
* User goes to notebook gradeable B, `initializePopupTimer` fires

Assuming the above, on _any_ page load of the notebook gradeable, then `sessionStorage.setItem('show_popup_submit', 'yes')` happens resetting any previous value. Given this, I'm saying you can add a new `let showPopup = true;` variable above this function, and just use that to track the state and avoid `sessionStorage`.

Reply:
but then wouldn't showPopup be set to true on every reload even if the user selects "don't show me again"

Reply:
So if someone chooses don't remind me again and then they make a submission, the variable will be reset when the page reloads and the popup will show again

Reply:
Disregard previous comments, I switched it to use local storage so that page reloads don't reset the value and different gradeables can have different values.